velox/common/base/test_utils/SuccinctPrintTest.cpp

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/SuccinctPrintUtil.h"
+
+#include <gtest/gtest.h>
+
+namespace facebook {
+namespace velox {
+namespace printer {
+
+TEST(SuccintPrintTest, testSuccintNanos) {
+  EXPECT_EQ(succinctNanos(123), "123ns");
+  EXPECT_EQ(succinctNanos(1'000), "1.0000us");
+  EXPECT_EQ(succinctNanos(1'234), "1.2340us");
+  EXPECT_EQ(succinctNanos(123'456), "123.4560us");
```

Comment:
Any particular reason for default precision of 4. It feels like 2 would be enough.

Reply:
Sure. Will use 2.

Reply:
We often do see queries running for much longer than seconds in CPU time. It would be nice to go at least to minutes, but days would give us 100% coverage.